Fixup: invert behavior of PropertyGraph::NeedsInference()

scober opened the pull request:
PropertyGraph::NeedsTypeInference() passes through the logic of
RDG::IsEntityTypeIDsOutsideProperties(), but the name suggests that it
will return the inverse value of what the RDG function returns. Invert
the logic and keep the name.

ddn0 commented on name-service/deploy/docker.sh:
```diff
@@ -0,0 +1,103 @@
+trap "{ kill_containers 'kns_*'; exit 255; }" EXIT
+kill_containers 'kns_*'
+
+readonly BASE=$(cd $(dirname "$0"); pwd)
```
TIL `readonly`

ddn0 commented on name-service/deploy/docker.sh:
```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# deploy Katana Name Service (KNS)
```
FWIW, I'm happy to not use bash scripts for everything. There just happens to be n+1 ways to babysit a set of processes and I figured this would be the easiest for people to follow and repurpose.

witchel replied:
Yeah, I looked at https://amoffat.github.io/sh/ one day while daydreaming about replacing all bash scripts with python.

tylershunt replied:
`sh` is great! I used it in `./scripts/cli_ctl`

tylershunt replied:
I ended up just re-writing it in python, lmk what you think!

witchel replied:
f strings, real programming language, what's not to like?  Maybe job control is a little clunkier, but this will be much easier to maintain.

I didn't even know about click.  Looks sharp.

tylershunt replied:
click is the best available python cli tool/arg parser imo.

ddn0 commented on name-service/name-server/name-server.go:
```diff
@@ -7,6 +7,7 @@ import (
 	"net/http"
 	"os"
 	"strconv"
+  "strings"
```
gofmt

ddn0 commented on name-service/deploy/docker/nameserver:
```diff
@@ -0,0 +1,7 @@
+FROM golang:1.15
```
not necessary, but in case you wanted to make a minimal deployed image you could do a multistage build.

I'll probably get the exact syntax wrong but something like:

```dockerfile
FROM golang:1.15 as build

WORKDIR /go/src/name-server

COPY ./name-server .
RUN go get -v ./...
RUN go install -v ./...

FROM debian:XY-slim

COPY --from=build /go/bin/nameserver /usr/local/bin/nameserver
CMD /usr/local/bin/nameserver
```

tylershunt commented:
I'm still debugging CI (if anyone has a better suggestion for CI based service management than `tmux` lmk, but marking this as ready to communicate my intent to merge soon :D.

witchel reviewed (approved):
Operational maturity, here we come.

ddn0 commented:
> if anyone has a better suggestion for CI based service management than tmux

Oh there are tons of these things, but in the absence of any other requirements, something that is easy to understand and replace if needed like `tmux` is probably the best balance to strike.

The next inflection point will be when we pick a packaging format (deb, docker, VM images, etc) and the CI does a test deployment using those artifacts.

Commit:
Fixup: invert behavior of PropertyGraph::NeedsInference()

PropertyGraph::NeedsTypeInference() passes through the logic of
RDG::IsEntityTypeIDsOutsideProperties(), but the name suggests that it
will return the inverse value of what the RDG function returns. Invert
the logic and keep the name.
